### PR TITLE
Browse All Event ViewModel

### PR DESCRIPTION
**Pull request by wisdomseeker2001**
Hi, this is fairly simple to understand when you look at code.

Decided to show following in the list:
Photo, Name, Date 

if you objections, lets talk about it,  all event apps usually have:
Photo, Name, Date  + Location as preview 

I suggest we keep it simple, description should only be relevant once someone clicks on it and reads more, but first they should see the date to see if they can make it, especially if we are gonna sort the vents by date by default (to be discussed)

### CODE REVIEW

File: app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt
Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt:1]: Fix Line 31, should be: _uiState.value = _uiState.value + eventListView

File: app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+class AllEventOverview(private val db: FirebaseFirestore = Firebase.firestore): ViewModel() {
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt:14]: This is not a good name. What is "all events"? `ViewModel` should be in the name.

File: app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    private val _uiState = MutableStateFlow<List<EventListingUiState>>(emptyList())
+    val uiState: StateFlow<List<EventListingUiState>> = _uiState
+
+    fun getEventData(): Unit {
+                db.collection("EVENT")
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt:20]: nit: `event` should be lowercase.

File: app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                    .addOnSuccessListener { documents ->
+                        for (document in documents) {
+                            val startInstant: Instant = document.getTimestamp("Start")!!.toDate().toInstant()
+
+                            val eventListView = EventListingUiState(
+                                document.getString("Name") ?: "",
+                                       document.getString("Photo") ?: "",
+                                        LocalDateTime.ofInstant(startInstant, ZoneId.systemDefault())
+                            )
+                            _uiState.value + eventListView
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt:31]: What is this line doing? I think this is the line Nathaniel said is wrong.

File: app/src/main/java/com/github/se/eventradar/model/event/Event.kt [hunk 4]
```diff
@@ -11,13 +11,19 @@ enum class EventCategory {
...
+    val start: LocalDateTime,
+    val end: LocalDateTime,
+    val location: Location,
     val description: String,
-    val date: LocalDate,
-    val time: LocalTime,
-    val category: EventCategory
-    // TODO: Add 'location' attribute once the Map feature is implemented
+    val ticket: Ticket,
+    val contact: String,
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/model/event/Event.kt:24]: suggestion: I think this should be a reference to the uid of the Host, not just a contact email string.

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt [hunk 5]
```diff
@@ -44,41 +44,57 @@ import androidx.compose.ui.unit.dp
...
+import com.github.se.eventradar.model.event.Ticket
 import com.github.se.eventradar.ui.BottomNavigationMenu
 import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
-import java.time.LocalDate
-import java.time.LocalTime
+import java.time.LocalDateTime
 
 @Composable
-fun HomeScreen() {
+fun HomeScreen() { // viewModel: EventViewModel = viewModel() + Navigation
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:56]: Has this been synced with main? There should be a navigation actions and view model in the parameters.

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt [hunk 6]
```diff
@@ -207,7 +223,7 @@ fun EventCard(event: Event) {
                       .weight(1f),
               verticalArrangement = Arrangement.Center) {
                 Text(
-                    text = event.name,
+                    text = event.eventName,
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:226]: nit: `name` would be a better variable name for this.

**Review by jgeaso1266 — CHANGES_REQUESTED** [review]
A few suggestions and name change ideas.

**Comment by christinehahaa** [comment]
This PR request has been closed and replaced with PR request #125 for the following reasons:
- ViewModel UI state was modified to allow more flexibility for getting a singular event, a list of filtered events, or a list of all events
- ViewModel extracts more data, which can be used for other functionalities
- It is the UI that chooses what is/isn't displayed.
   - Joyti's comments regarding what is displayed in the list is still relevant, but should be noted for the devop of the UI.

**Pull request closed without merging**